### PR TITLE
Type safe comparison of strings only, in libraries/cms

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
- Type safe string comparison of strings only, in libraries/cms

This PR is part of a set to try to separate some of the changes done in one of my previous batch PR's for the libraries/cms directory, which is still on hold (#12171).
Once the new set is merged completely, it will hopefully reduce the changes in that PR, so it can be reviewed easier and finally be merged.

The changes in this PR should be very easy to review, as they are only string and string type comparisons. In hope that this will get merged quickly. ;)

Note: Don't bother if some possible changes are missing or could be differently written. They are probably in the batch PR , that this one references. As soon as this set of sub PR's is merged, the batch PR will have its conflicts resolved and should be a lot easier to review and finally get merged.

### Testing Instructions

None, should not change behavior

### Documentation Changes Required

None.

### CODE REVIEW

**Review by andrepereiradasilva — APPROVED** [review]

**Comment by frankmayer** [comment]
Conflicts resolved...

**Comment by frankmayer** [comment]
Conflicts resolved. Pls check and merge.

File: libraries/cms/application/site.php [hunk 1]
```diff
@@ -741,7 +741,7 @@ protected function render()
 				$template = $this->getTemplate(true);
 				$file     = $this->input->get('tmpl', 'index');
 
-				if (!$this->get('offline') && ($file == 'offline'))
+				if (!$this->get('offline') && ($file === 'offline'))
```
Comment (Quy) [inline libraries/cms/application/site.php]: Remove parentheses `($file === 'offline')`

File: libraries/cms/form/field/limitbox.php [hunk 2]
```diff
@@ -80,7 +80,7 @@ protected function getOptions()
 			asort($limits);
 
 			// Add an option to show all?
-			$showAll = isset($this->element['showall']) ? ($this->element['showall'] == 'true') : true;
+			$showAll = isset($this->element['showall']) ? ($this->element['showall'] === 'true') : true;
```
Comment (Quy) [inline libraries/cms/form/field/limitbox.php]: Remove parentheses `($this->element['showall'] === 'true')`

File: libraries/cms/form/field/tag.php [hunk 3]
```diff
@@ -223,7 +223,7 @@ public function isNested()
 		if (is_null($this->isNested))
 		{
 			// If mode="nested" || ( mode not set & config = nested )
-			if ((isset($this->element['mode']) && $this->element['mode'] == 'nested')
+			if ((isset($this->element['mode']) && $this->element['mode'] === 'nested')
```
Comment (Quy) [inline libraries/cms/form/field/tag.php]: Remove parentheses `(isset($this->element['mode'])`

File: libraries/cms/html/grid.php [hunk 4]
```diff
@@ -76,22 +76,22 @@ public static function sort($title, $order, $direction = 'asc', $selected = '',
...
+		$index = (int) ($direction === 'desc');
 
 		if ($order != $selected)
 		{
 			$direction = $new_direction;
 		}
 		else
 		{
-			$direction = ($direction == 'desc') ? 'asc' : 'desc';
+			$direction = ($direction === 'desc') ? 'asc' : 'desc';
```
Comment (Quy) [inline libraries/cms/html/grid.php]: Remove parentheses `($direction === 'desc')`

File: libraries/cms/html/searchtools.php [hunk 5]
```diff
@@ -139,15 +139,15 @@ public static function sort($title, $order, $direction = 'asc', $selected = 0, $
...
+		$index = (int) ($direction === 'desc');
 
 		if ($order != $selected)
 		{
 			$direction = $new_direction;
 		}
 		else
 		{
-			$direction = ($direction == 'desc') ? 'asc' : 'desc';
+			$direction = ($direction === 'desc') ? 'asc' : 'desc';
```
Comment (Quy) [inline libraries/cms/html/searchtools.php]: Remove parentheses `($direction === 'desc')`

File: libraries/cms/installer/adapter/plugin.php [hunk 6]
```diff
@@ -359,14 +359,14 @@ protected function setupInstallPaths()
...
-		if ($this->route == 'discover_install')
+		if ($this->route === 'discover_install')
 		{
 			$manifest_details = JInstaller::parseXMLInstallFile($this->parent->getPath('manifest'));
 
 			$this->extension->manifest_cache = json_encode($manifest_details);
 			$this->extension->state = 0;
 			$this->extension->name = $manifest_details['name'];
-			$this->extension->enabled = ('editors' == $this->extension->folder) ? 1 : 0;
+			$this->extension->enabled = ('editors' === $this->extension->folder) ? 1 : 0;
```
Comment (Quy) [inline libraries/cms/installer/adapter/plugin.php]: Remove parentheses `('editors' === $this->extension->folder)`

File: libraries/cms/installer/installer.php [hunk 7]
```diff
@@ -1330,7 +1330,7 @@ public function parseFiles(SimpleXMLElement $element, $cid = 0, $oldFiles = null
 			$path['dest'] = $destination . '/' . $file;
 
 			// Is this path a file or folder?
-			$path['type'] = ($file->getName() == 'folder') ? 'folder' : 'file';
+			$path['type'] = ($file->getName() === 'folder') ? 'folder' : 'file';
```
Comment (Quy) [inline libraries/cms/installer/installer.php]: Remove parentheses

File: libraries/cms/installer/installer.php [hunk 8]
```diff
@@ -1527,7 +1527,7 @@ public function parseMedia(SimpleXMLElement $element, $cid = 0)
...
```
Comment (Quy) [inline libraries/cms/installer/installer.php]: Remove parentheses

File: libraries/cms/module/helper.php [hunk 9]
```diff
@@ -295,7 +295,7 @@ public static function getLayoutPath($module, $layout = 'default')
 		{
 			// Get the template and file name from the string
 			$temp = explode(':', $layout);
-			$template = ($temp[0] == '_') ? $template : $temp[0];
+			$template = ($temp[0] === '_') ? $template : $temp[0];
```
Comment (Quy) [inline libraries/cms/module/helper.php]: Remove parentheses

File: libraries/cms/plugin/helper.php [hunk 10]
```diff
@@ -44,7 +44,7 @@ public static function getLayoutPath($type, $name, $layout = 'default')
 		{
 			// Get the template and file name from the string
 			$temp = explode(':', $layout);
-			$template = ($temp[0] == '_') ? $template : $temp[0];
+			$template = ($temp[0] === '_') ? $template : $temp[0];
```
Comment (Quy) [inline libraries/cms/plugin/helper.php]: Remove parentheses

File: libraries/cms/application/cms.php [hunk 11]
```diff
@@ -272,7 +272,7 @@ public function execute()
 		}
 
 		// If gzip compression is enabled in configuration and the server is compliant, compress the output.
-		if ($this->get('gzip') && !ini_get('zlib.output_compression') && (ini_get('output_handler') != 'ob_gzhandler'))
+		if ($this->get('gzip') && !ini_get('zlib.output_compression') && (ini_get('output_handler') !== 'ob_gzhandler'))
```
Comment (Quy) [inline libraries/cms/application/cms.php]: Remove parentheses

File: libraries/cms/application/site.php (hunk 1 above)
Comment (Quy) [inline libraries/cms/application/site.php]: Remove parentheses

File: libraries/cms/form/field/limitbox.php (hunk 2 above)
Comment (Quy) [inline libraries/cms/form/field/limitbox.php]: Remove parentheses

File: libraries/cms/form/field/tag.php (hunk 3 above)
Comment (Quy) [inline libraries/cms/form/field/tag.php]: Remove parentheses

File: libraries/cms/html/grid.php (hunk 4 above)
Comment (Quy) [inline libraries/cms/html/grid.php]: Remove parentheses

File: libraries/cms/html/searchtools.php (hunk 5 above)
Comment (Quy) [inline libraries/cms/html/searchtools.php]: Remove parentheses

File: libraries/cms/installer/adapter/plugin.php (hunk 6 above)
Comment (Quy) [inline libraries/cms/installer/adapter/plugin.php]: Remove parentheses

File: libraries/cms/installer/installer.php [hunk 12]
```diff
@@ -954,15 +954,15 @@ public function parseSQLFiles($element)
 		// Get the name of the sql file to process
 		foreach ($element->children() as $file)
 		{
-			$fCharset = (strtolower($file->attributes()->charset) == 'utf8') ? 'utf8' : '';
+			$fCharset = (strtolower($file->attributes()->charset) === 'utf8') ? 'utf8' : '';
```
Comment (Quy) [inline libraries/cms/installer/installer.php]: Remove parentheses

File: libraries/cms/installer/installer.php (hunk 7 above)
Comment (Quy) [inline libraries/cms/installer/installer.php]: Remove parentheses

File: libraries/cms/installer/installer.php (hunk 8 above)
Comment (Quy) [inline libraries/cms/installer/installer.php]: Remove parentheses

File: libraries/cms/module/helper.php (hunk 9 above)
Comment (Quy) [inline libraries/cms/module/helper.php]: Remove parentheses

File: libraries/cms/plugin/helper.php (hunk 10 above)
Comment (Quy) [inline libraries/cms/plugin/helper.php]: Remove parentheses

File: libraries/cms/module/helper.php [hunk 13]
```diff
@@ -295,7 +295,7 @@ public static function getLayoutPath($module, $layout = 'default')
...
 			$layout = $temp[1];
 			$defaultLayout = ($temp[1]) ? $temp[1] : 'default';
```
Comment (Quy) [inline libraries/cms/module/helper.php]: Remove parentheses

File: libraries/cms/plugin/helper.php [hunk 14]
```diff
@@ -44,7 +44,7 @@ public static function getLayoutPath($type, $name, $layout = 'default')
...
 			$layout = $temp[1];
 			$defaultLayout = ($temp[1]) ? $temp[1] : 'default';
```
Comment (Quy) [inline libraries/cms/plugin/helper.php]: Remove parentheses

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on bae19d7d22ff0aa5030e1c96b9e5a3e68fe6cc35<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13275">issues.joomla.org/tracker/joomla-cms/13275</a>.</sub>

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on bae19d7d22ff0aa5030e1c96b9e5a3e68fe6cc35<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13275">issues.joomla.org/tracker/joomla-cms/13275</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Pull request merged**